pandas/tools/merge.py

```diff
@@ -193,6 +193,10 @@ def __init__(self, left, right, how='inner', on=None,
                 'can not merge DataFrame with instance of '
                 'type {0}'.format(type(right)))
 
+        # prevent merging between different levels
+        if left.columns.nlevels != right.columns.nlevels:
+            raise ValueError('can not merge between different levels')
```

Comment:
can you make this even more expressive, e.g. indicate which/how many on left & right.